auth/authpb/auth.proto

```diff
@@ -16,7 +16,21 @@ message User {
   int64 tombstone = 3;
 }
 
+// Permission is a single entity
+message Permission {
+  bytes path = 1;
+
+  enum PermissionType {
+    READ = 0;
+    WRITE = 1;
+  }
+  int32 permissionBitmap = 2;
+}
+
 // Role is a single entry in the bucket authRoles
 message Role {
-  bytes name = 2;
+  bytes name = 1;
+
+  repeated Permission pathPermission = 2;
```

Comment:
@xiang90 @mitake This should support ranges if possible. Otherwise KV requests won't have a direct mapping to auth permissions; it seems incomplete. 

The interval tree code in pkg/adt should simplify handling any range queries the auth code may need.